plugins/terraform-backend/src/lib/index.ts

```diff
@@ -1,59 +1,26 @@
 import axios from 'axios';
 import {
-  TerraformEntity,
+  ResponseRun,
   TerraformResponse,
   TerraformRun,
   TerraformWorkspace,
 } from './types';
+import { terraformRun2ResponseRun } from '../terraformRun2ResponseRun';
 
-const TF_BASE_URL = 'https://app.terraform.io/api/v2';
-
-const findEntityById = (entities: TerraformEntity[], id: string) =>
-  entities.find(e => e.id === id);
-
-const getPlanDetails = (plans: TerraformEntity[], planId?: string) => {
-  if (!planId) return null;
-
-  const plan = findEntityById(plans, planId);
-
-  if (!plan) return null;
-
-  return { logs: plan.attributes['log-read-url'] };
-};
-
-const getUserDetails = (users: TerraformEntity[], userId?: string) => {
-  if (!userId) return null;
-
-  const user = findEntityById(users, userId);
-
-  if (!user) return null;
-
-  return {
-    name: user.attributes.username,
-    avatar: user.attributes['avatar-url'],
-  };
-};
+export const TF_BASE_URL = 'https://app.terraform.io/api/v2';
 
 export const getRunsForWorkspace = async (
   token: string,
   workspaceId: string,
-) => {
+): Promise<ResponseRun[]> => {
   const res = await axios.get<TerraformResponse<TerraformRun[]>>(
     `${TF_BASE_URL}/workspaces/${workspaceId}/runs?include=created_by,plan`,
     { headers: { Authorization: `Bearer ${token}` } },
   );
 
-  return res.data.data.map(r => ({
-    id: r.id,
-    message: r.attributes.message,
-    status: r.attributes.status,
-    createdAt: r.attributes['created-at'],
-    confirmedBy: getUserDetails(
-      res.data.included,
-      r.relationships['confirmed-by']?.data.id,
-    ),
-    plan: getPlanDetails(res.data.included, r.relationships.plan?.data.id),
-  }));
+  return res.data.data.map<ResponseRun>(singleRun =>
```

Comment:
is the `<ResponseRun>` type required here? i'd have thought it would already know the type as we type the axios call with `<TerraformResponse<TerraformRun[]>>` above

Reply:
Thank you for the suggestion, I'll remove it as it is redundant.